sci-libs/parmetis/parmetis-4.0.3-r1.ebuild

```diff
@@ -22,16 +22,14 @@ KEYWORDS="amd64 arm64 x86 ~amd64-linux ~x86-linux"
 IUSE="doc double-precision examples int64 mpi openmp pcre static-libs"
 RESTRICT="mirror bindist"
 
-DEPEND="mpi? ( virtual/mpi )"
+DEPEND="mpi? ( virtual/mpi )
+	pcre? ( dev-libs/libpcre:= )"
 RDEPEND="${DEPEND}
 	!<sci-libs/metis-5"
 
 pkg_setup() {
 	if use openmp; then
-		if [[ $(tc-getCC)$ == *gcc* ]] && ! tc-has-openmp; then
-			ewarn "You are using gcc but openmp is not available"
-			die "Need an OpenMP capable compiler"
-		fi
+		tc-check-openmp
 	fi
```

Comment:
change the whole block to
`[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp`